shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -485,6 +487,28 @@ - (BOOL)createShell:(NSString*)entrypoint libraryURI:(NSString*)libraryURI {
                                       _threadHost.ui_thread->GetTaskRunner(),          // ui
                                       _threadHost.io_thread->GetTaskRunner()           // io
     );
+    // Create the shell with async mode . This is a not blocking operation.
+    if (oc_async_init_block) {
+      oc_async_init_block = [oc_async_init_block copy];
+      FlutterEngine* selfRef = [self retain];
+      flutter::Shell::CreateAsync(
+          [oc_async_init_block, selfRef](bool success, std::unique_ptr<flutter::Shell> shell) {
+            if (!success) {
+              FML_LOG(ERROR) << "Could not start a shell FlutterEngine on asyncInitMode";
+              oc_async_init_block(false);
+            } else {
+              selfRef->_shell = std::move(shell);
```

Comment:
What's to stop someone from invoking `sendOnChannel:message:` before `_shell` is set?  How would they even know when it was safe to call it?